source/MaterialXContrib/MaterialXNode/CreateMaterialXNodeCmd.cpp

```diff
@@ -116,78 +116,94 @@ MStatus CreateMaterialXNodeCmd::doIt( const MArgList &args )
             CHECK_MSTATUS(argData.getFlagArgument(kElementFlag, 0, elementPath));
         }
 
-        std::unique_ptr<MaterialXData> materialXData{
-            new MaterialXData(document,
-                              elementPath.asChar(),
-                              Plugin::instance().getLibrarySearchPath())
-        };
-
-        MString ogsXmlFileName;
-        if (parser.isFlagSet(kOgsXmlFlag))
+        // Determine what renderables will be used to create fragment shaders and Maya nodes
+        std::string desiredElementPath(elementPath.asChar());
+        std::vector<mx::TypedElementPtr> renderableElements{
+            MaterialXMaya::getRenderableElements(document, desiredElementPath) };
+
+        const MaterialX::FileSearchPath& searchPath = Plugin::instance().getLibrarySearchPath();
+        MStringArray nodeNames;
+        for (auto renderableElement : renderableElements)
```

Comment:
Then we iterate over all renderables we want to consider. I don't think we need a "only first renderable" item as this makes it inconsistent with the viewer and node editor. Also this should allow us to scan for looks and perform material assignments knowing the materials in the document have been created. A different task.

Reply:
The body of this loop could be a lambda or a function, to make it reusable without requiring a vector.

Reply:
It's not really reusable code outside this command so I'm not sure I see the point?

Reply:
I mean, if the getRenderableElements() is split in two as I'm suggesting below and we don't always have a vector, then a lambda or a function would be a way to reuse this code.

Reply:
I made the change. I was able to localize it be bit more but still a number of args to pass over.